mmv1/third_party/terraform/services/compute/resource_compute_instance_test.go.tmpl

```diff
@@ -11846,4 +12056,508 @@ resource "google_compute_instance" "foobar" {
 	}
 }
 `, context)
+}
+
+func testAccComputeInstance_instanceEncryption_SelfLinkServiceAccount(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+data "google_compute_default_service_account" "default" {}
+
+data "google_project" "default" {}
+
+resource "google_project_iam_member" "default-service-agent" {
```

Comment:
You'll want to use a bootstrapped permission instead of adding these, so that the permission is added and persisted across tests, and not removed after each test (this is mainly important because other tests could be using this same permission).

It would be something like this in your test function:

```
	acctest.BootstrapIamMembers(t, []acctest.IamMember{
		{
			Member: "serviceAccount:${project_number}-compute@developer.gserviceaccount.com",
			Role:   "roles/cloudkms.cryptoKeyEncrypterDecrypter",
		},
	})
```